### PR TITLE
Make public interfaces implementable

### LINKED ISSUE #183: Recent changes made imposible to have custom implementations for common interfaces
Having private interface `token`:
https://github.com/scylladb/gocql/blob/7f7905db2c37b557fd406596b3f7c456ed3ea1d3/token.go#L27-L30
And `partitioner`:
https://github.com/scylladb/gocql/blob/7f7905db2c37b557fd406596b3f7c456ed3ea1d3/token.go#L20-L24

In public interfaces totaly breaks possibility of their custom implementation.
Affected interfaces:
1. `SelectedHost`:
https://github.com/scylladb/gocql/blob/7f7905db2c37b557fd406596b3f7c456ed3ea1d3/policies.go#L324-L328

2. `ExecutableQuery`:
https://github.com/scylladb/gocql/blob/3c32c6cd75e529433ed0885d8c04ec4c96cf22a9/query_executor.go#L9-L28

3. `NextHost`, because of references to `SelectedHost`
https://github.com/scylladb/gocql/blob/7f7905db2c37b557fd406596b3f7c456ed3ea1d3/policies.go#L346

4. `HostSelectionPolicy`, because of reference to `NextHost`

### PR DESCRIPTION
**Pull request by dkropachev**
currently private interface `token` directly or indirectly present in lot's of interfaces and therefore makes them unimplementable: `SelectedHost`, `ExecutableQuery`, `HostSelectionPolicy`, `NextHost`

To fix it we need to make all parts of interfaces be public
Closes #183

### CODE REVIEW

File: host_source.go [hunk 1]
```diff
@@ -435,10 +435,10 @@ func (h *HostInfo) Hostname() string {
...
 
 func (h *HostInfo) ConnectAddressAndPort() string {
-        h.mu.Lock()
-        defer h.mu.Unlock()
-        addr, _ := h.connectAddressLocked()
-        return net.JoinHostPort(addr.String(), strconv.Itoa(h.port))
+	h.mu.Lock()
+	defer h.mu.Unlock()
+	addr, _ := h.connectAddressLocked()
+	return net.JoinHostPort(addr.String(), strconv.Itoa(h.port))
```
Comment (sylwiaszunejko) [inline host_source.go:441]: Not related to the PR

File: metadata_scylla.go [hunk 2]
```diff
@@ -244,7 +244,7 @@ type schemaDescriber struct {
 	session *Session
 	mu      sync.Mutex
 
-	cache        map[string]*KeyspaceMetadata
+	cache map[string]*KeyspaceMetadata
```
Comment (sylwiaszunejko) [inline metadata_scylla.go:247]: Not related to the PR

File: scylla.go [hunk 3]
```diff
@@ -378,13 +378,13 @@ func (p *scyllaConnPicker) Pick(t token, keyspace string, table string) *Conn {
 		conn.mu.Lock()
 		if conn.tabletsRoutingV1 {
 			tablets := conn.session.getTablets()
-	
+
```
Comment (sylwiaszunejko) [inline scylla.go:381]: As before

File: scylla.go [hunk 4]
```diff
@@ -415,15 +415,15 @@ func (p *scyllaConnPicker) maybeReplaceWithLessBusyConnection(c *Conn) *Conn {
...
+	if alternative == nil || alternative.AvailableStreams()*120 > c.AvailableStreams()*100 {
 		return c
 	} else {
 		return alternative
 	}
 }
 
 func isHeavyLoaded(c *Conn) bool {
-    return c.streams.NumStreams / 2 > c.AvailableStreams();
+	return c.streams.NumStreams/2 > c.AvailableStreams()
```
Comment (sylwiaszunejko) [inline scylla.go:426]: Not related to the PR

File: session.go [hunk 5]
```diff
@@ -2171,12 +2171,12 @@ type routingKeyInfoLRU struct {
...
 
 type routingKeyInfo struct {
-	indexes  []int
-	types    []TypeInfo
-	keyspace string
-	table    string
+	indexes     []int
+	types       []TypeInfo
+	keyspace    string
+	table       string
```
Comment (sylwiaszunejko) [inline session.go:2177]: Not related

**Review by sylwiaszunejko — CHANGES_REQUESTED** [review]

File: scylla.go (hunk 4 above)
Comment (dkropachev) [reply scylla.go]: removed

File: scylla.go (hunk 3 above)
Comment (dkropachev) [reply scylla.go]: removed

File: metadata_scylla.go (hunk 2 above)
Comment (dkropachev) [reply metadata_scylla.go]: removed

File: session.go (hunk 5 above)
Comment (dkropachev) [reply session.go]: removed

**Pull request merged**